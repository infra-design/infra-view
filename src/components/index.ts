export * from './Absolute'
export * from './Container'
export * from './Fixed'
export * from './Grid'
export * from './View'
