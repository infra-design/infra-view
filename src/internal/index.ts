export { jsx } from '@emotion/react'
