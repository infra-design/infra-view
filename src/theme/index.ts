export * from './baseTheme'
export * from './createTheme'
export * from './useTheme'
