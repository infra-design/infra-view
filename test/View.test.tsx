import { render } from '@testing-library/react'
import React from 'react'
import { View } from '../src'

test('<View />', async () => {
  const view = render(<View />)

  expect(view.container).toMatchSnapshot()
})

test('<View><View></View></View>', async () => {
  const view = render(
    <View>
      <View></View>
    </View>,
  )

  expect(view.container).toMatchSnapshot()
})

test(`<View as='button'>Button</View>`, () => {
  const view = render(<View as='button'>Button</View>)

  expect(view.container).toMatchSnapshot()
})

test(`<View position='absolute' />`, () => {
  const view = render(<View position='absolute' />)

  expect(view.container).toMatchSnapshot()
})

test(`<View margin={10} />`, () => {
  const view = render(<View margin={10} />)

  expect(view.container).toMatchSnapshot()
})

test(`<View backgroundImageUrl='https://foo/bar.jpg' />`, () => {
  const view = render(<View backgroundImageUrl='https://foo/bar.jpg' />)

  expect(view.container).toMatchSnapshot()
})
