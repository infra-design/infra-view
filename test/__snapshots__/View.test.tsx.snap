// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<View /> 1`] = `
<div>
  <div
    class="emotion-0"
  />
</div>
`;

exports[`<View as='button'>Button</View> 1`] = `
<div>
  <button
    class="emotion-0"
  >
    Button
  </button>
</div>
`;

exports[`<View backgroundImageUrl='https://foo/bar.jpg' /> 1`] = `
.emotion-0 {
  background-image: url("https://foo/bar.jpg");
}

<div>
  <div
    class="emotion-0"
  />
</div>
`;

exports[`<View margin={10} /> 1`] = `
.emotion-0 {
  margin: 10px;
}

<div>
  <div
    class="emotion-0"
  />
</div>
`;

exports[`<View position='absolute' /> 1`] = `
.emotion-0 {
  position: absolute;
}

<div>
  <div
    class="emotion-0"
  />
</div>
`;

exports[`<View><View></View></View> 1`] = `
<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-0"
    />
  </div>
</div>
`;
